infrastructure/data_access/http/synthetics_repo.py

```diff
@@ -115,3 +110,19 @@ def scale_us_to_ms(val: str) -> MetricValue:
 def scale_to_percents(val: str) -> MetricValue:
     # scale 0..1 -> 0..100
     return MetricValue(float(val) * 100.0)
+
+
+def make_internal_agents(agents, agent_ids) -> Agents:
```

Comment:
The code is correct, but inefficient. There is no need to retrieve list of all agents every time test health data are retrieved. Number of available Kentik agents is only going to grow, so the overhead will grow over time. I suggest to get list of all agents when the app starts and after that only when the test config contains an agent ID not available in the list of agents (i.e. it refers to an agent that has been added since the last time the list of agents was retrieved). The `SyntheticsRepo` class should hold an instance of `Agents` class and update it as necessary. This function should be turned into an internal method `SyntheticsRepo` and only pick agent data from the instance of `Agents` owned by the class as opposed to constructing the list of agents every time.

Reply:
I'll create a separate PR with optimization.